src/main/java/org/kiwiproject/elk/ElkAppenderFactory.java

```diff
@@ -0,0 +1,73 @@
+package org.kiwiproject.elk;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotBlank;
+import static org.kiwiproject.collect.KiwiMaps.isNotNullOrEmpty;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.dropwizard.logging.AbstractAppenderFactory;
+import io.dropwizard.logging.async.AsyncAppenderFactory;
+import io.dropwizard.logging.filter.LevelFilterFactory;
+import io.dropwizard.logging.layout.LayoutFactory;
+import lombok.Getter;
+import lombok.Setter;
+import net.logstash.logback.appender.LogstashUdpSocketAppender;
+import net.logstash.logback.layout.LogstashLayout;
+import org.kiwiproject.json.JsonHelper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Getter
+@Setter
+@JsonTypeName("elk")
+public class ElkAppenderFactory extends AbstractAppenderFactory<ILoggingEvent> {
```

Comment:
Since this is public, should we add javdocs?

Reply:
This is meant to be used from the config (see the README) so I'm not going to for now